File: app/src/main/java/cenergy/central/com/pwb_store/model/ProductDetailImage.java
package cenergy.central.com.pwb_store.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by napabhat on 10/25/2016 AD.
 */

public class ProductDetailImage implements IViewType, Parcelable {

    public static final Creator<ProductDetailImage> CREATOR = new Creator<ProductDetailImage>() {
        @Override
        public ProductDetailImage createFromParcel(Parcel in) {
            return new ProductDetailImage(in);
        }

        @Override
        public ProductDetailImage[] newArray(int size) {
            return new ProductDetailImage[size];
        }
    };
    private int viewTypeId;
    private int total;
    private ArrayList<ProductDetailImageItem> mProductDetailImageItems;
    private ProductDetailImageItem selectedProductDetailImageItem;

    public ProductDetailImage(int total, ArrayList<ProductDetailImageItem> productDetailImageItems) {
        this.total = total;
        this.mProductDetailImageItems = productDetailImageItems;

    }

    protected ProductDetailImage(Parcel in) {
        viewTypeId = in.readInt();
        total = in.readInt();
        List<ProductDetailImage> mProductDetailImageItems = new ArrayList<>();
        in.readList(mProductDetailImageItems, ProductDetailImage.class.getClassLoader());
        selectedProductDetailImageItem = in.readParcelable(ProductDetailImageItem.class.getClassLoader());
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(viewTypeId);
        dest.writeInt(total);
        dest.writeList(mProductDetailImageItems);
        dest.writeParcelable(selectedProductDetailImageItem, flags);
    }

    @Override
    public int getViewTypeId() {
        return viewTypeId;
    }

    @Override
    public void setViewTypeId(int id) {
        this.viewTypeId = id;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }

    public ArrayList<ProductDetailImageItem> getProductDetailImageItems() {
        return mProductDetailImageItems;
    }

    public void setProductDetailImageItems(ArrayList<ProductDetailImageItem> productDetailImageItems) {
        mProductDetailImageItems = productDetailImageItems;
    }


    public boolean isSelectedProductDetailImageItemAvailable() {
        return selectedProductDetailImageItem != null;
    }

    public ProductDetailImageItem getSelectedProductDetailImageItem() {
        return selectedProductDetailImageItem;
    }

    public void setSelectedProductDetailImageItem(ProductDetailImageItem selectedProductDetailImageItem) {
        this.selectedProductDetailImageItem = selectedProductDetailImageItem;
    }
}

File: app/src/main/java/cenergy/central/com/pwb_store/model/ProductDetailImage.kt
package cenergy.central.com.pwb_store.model

import android.os.Parcelable
import io.realm.RealmList
import io.realm.RealmModel
import io.realm.annotations.RealmClass
import kotlinx.android.parcel.Parcelize
import kotlinx.android.parcel.WriteWith

@Parcelize
@RealmClass
open class ProductDetailImage(
        var total: Int = 0,
        var productDetailImageItems: @WriteWith<ProductDetailImageRealmListParceler> RealmList<ProductDetailImageItem> = RealmList(),
        var viewTypeID: Int = 0,
        private var selectedProductDetailImageItem: ProductDetailImageItem? = null) : IViewType, Parcelable, RealmModel{

    override fun getViewTypeId(): Int {
        return viewTypeID
    }

    override fun setViewTypeId(id: Int) {
        viewTypeID = id
    }

    private fun validateProductDetailImageItemList() {
        var productDetailImageItem: ProductDetailImageItem
        for (i in productDetailImageItems.indices) {
            productDetailImageItem = productDetailImageItems[i]!!
            if (isSelectedProductDetailImageItemAvailable) {
                productDetailImageItem.isSelected = productDetailImageItem.imgUrl.equals(getSelectedProductDetailImageItem()!!.imgUrl, ignoreCase = true)
            }
        }
    }

    private val isSelectedProductDetailImageItemAvailable: Boolean
        get() = selectedProductDetailImageItem != null

    private fun getSelectedProductDetailImageItem(): ProductDetailImageItem? {
        return selectedProductDetailImageItem
    }

    fun setSelectedProductDetailImageItem(selectedProductDetailImageItem: ProductDetailImageItem?) {
        this.selectedProductDetailImageItem = selectedProductDetailImageItem
        validateProductDetailImageItemList()
    }
}